src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppFileAccessor.cs

```diff
@@ -47,20 +65,34 @@ private void DocData_Modifying(object sender, EventArgs e)
 
     private async Task<MyAppDocument?> TryGetMyAppFileAsync()
     {
-        if (_docData is null)
+        if (_docData?.Data is null || !File.Exists(_absolutePath))
```

Comment:
Checking if the file path exists sounds like a double validation here: we already know that if the DocData is null it means that there is no file, thus the file path doesn't exist neither.